### ISSUE TITLE
"0" in AddDuration causes a panic

### ISSUE THREAD

**Issue by mashedkeyboard** [karrick/tparse#2]
Hey, thanks for the awesome library :)

I noted that `tparse.AddDuration(time.Now(), "0")`, or any variant of it (`+0` or `-0`) causes a panic:

```panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/karrick/tparse.AddDuration(0xbfb454f2c9021b57, 0x31c5e, 0x5defe0, 0x50974c, 0x2, 0xbfb454f2c90219dd, 0x31aed, 0x5defe0, 0x0, 0x0)
        /home/curtispf/go/pkg/mod/github.com/karrick/tparse@v2.4.2+incompatible/tparse.go:196 +0x9ca
main.main()
+0x8c
exit status 2
```

However, `tparse.AddDuration(time.Now(), "-0months")` and the like work fine. Clearly 0 is the same as 0months, but I imagine this edge case of a valid unitless value hasn't been considered - it'd be great to have it accounted for! :smile:

**Comment by karrick** [comment]
I agree that the bug is there, and a stack trace is unacceptable. Side note, adding any number without a unit is an input error.

**Comment by mashedkeyboard** [comment]
Yeah, in any other context it makes no sense whatsoever - unless the library can read minds (which would be very impressive) unitless n for any n ≠ 0 is clearly nonsensical as a value.

**Comment by karrick** [comment]
Fixed by v2.8.2

**Comment by karrick** [comment]
@mashedkeyboard , please let me know if you find any other bizarre behavior, and I will be happy to address!
